### PR TITLE
Feature/integrate husky+prettier

### LINKED ISSUE #11: Integrate husky & prettier

### PR DESCRIPTION
**Pull request by SpiraTecGabelM**

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Ignore artifacts:
+/build
+/coverage
+/dist
```
Comment (EdgarKisman) [inline .prettierignore]: also add 

package-lock.json
node_modules/
.vscode/
.idea/
*.min.js
*.min.css
package-lock.json

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "trailingComma": "none",
+  "tabWidth": 2,
+  "semi": false,
+  "singleQuote": true
```
Comment (EdgarKisman) [inline .prettierrc]: also add 

trailingComma: es5
arrowParens: avoid

File: package.json [hunk 3]
```diff
@@ -7,13 +7,16 @@
...
+    "preview": "vite preview",
+    "commit": "pnpm exec prettier . --write && pnpm eslint \"{**/*.ts,**/*.tsx}\""
   },
   "dependencies": {
     "@types/xml2js": "0.4.14",
+    "eslint-config-prettier": "9.1.0",
     "fast-xml-parser": "4.3.2",
     "grommet": "2.35.0-alpha.1",
     "grommet-icons": "4.12.0",
+    "husky": "8.0.3",
```
Comment (EdgarKisman) [inline package.json]: i think husky is a dev dependency

File: package.json [hunk 4]
```diff
@@ -7,13 +7,16 @@
   "scripts": {
```
Comment (EdgarKisman) [inline package.json:7]: check which lib should be in dependency section and which in devDep.

File: src/main.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import ReactDOM from 'react-dom/client'
 import App from './App'
 import './index.css'
 
-ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
+ReactDOM.createRoot(document.getElementById('root')!).render(
```
Comment (EdgarKisman) [inline src/main.tsx]: kein ! benutzen

File: src/pages/login/LoginPage.tsx [hunk 6]
```diff
@@ -10,13 +10,11 @@ import {
...
 import { type Credentials } from '../../api/models'
 import { AuthContext } from '../../context/AuthContext'
-import isNil from 'lodash/isNil'
-import { Blank } from '../../utils/constants'
 
 const LoginPage = (): JSX.Element => {
   const [isLoading, setIsLoading] = useState<boolean>(false)
-  const [username, setUsername] = useState<string>(Blank)
-  const [password, setPassword] = useState<string>(Blank)
+  const [username, setUsername] = useState<string>('')
```
Comment (EdgarKisman) [inline src/pages/login/LoginPage.tsx]: improve

File: src/pages/login/LoginPage.tsx [hunk 7]
```diff
@@ -62,7 +64,7 @@ const LoginPage = (): JSX.Element => {
           />
         </Box>
       </Form>
-      {!isNil(error) && (
+      {error !== null && error !== undefined && (
```
Comment (EdgarKisman) [inline src/pages/login/LoginPage.tsx]: improve

**Comment by SpiraTecGabelM** [comment]
Done

**Review by EdgarKisman — APPROVED** [review]

File: .prettierignore (hunk 1 above)
Comment (SpiraTecGabelM) [reply .prettierignore]: Done

File: .prettierrc (hunk 2 above)
Comment (SpiraTecGabelM) [reply .prettierrc]: Done

File: package.json (hunk 4 above)
Comment (SpiraTecGabelM) [reply package.json:7]: done

File: package.json (hunk 3 above)
Comment (SpiraTecGabelM) [reply package.json]: done

File: src/pages/login/LoginPage.tsx (hunk 6 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.tsx]: done

File: src/pages/login/LoginPage.tsx (hunk 7 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.tsx]: done

File: src/main.tsx (hunk 5 above)
Comment (SpiraTecGabelM) [reply src/main.tsx]: done

**Pull request merged**